src/input/keyboard.c

```diff
@@ -608,11 +610,30 @@ keyboard_init(struct seat *seat)
 		wlr_log(WLR_ERROR, "Failed to create xkb keymap");
 	}
 	xkb_context_unref(context);
-	wlr_keyboard_set_repeat_info(kb, rc.repeat_rate, rc.repeat_delay);
+}
 
+void
+keyboard_configure(struct seat *seat, struct wlr_keyboard *kb, bool is_virtual)
+{
+	if (!is_virtual) {
+		set_layout(kb);
+	}
+	wlr_keyboard_set_repeat_info(kb, rc.repeat_rate, rc.repeat_delay);
 	keybind_update_keycodes(seat->server);
 }
 
+void
+keyboard_group_init(struct seat *seat)
+{
+	if (seat->keyboard_group) {
+		wlr_log(WLR_ERROR, "keyboard group already exists - noop");
+		return;
+	}
```

Comment:
IMHO this should either be an `assert()` or we don't need to log here.

Reply:
Done. `assert()` used.